storage/replica.go

```diff
@@ -843,19 +843,19 @@ func (r *Replica) Send(ctx context.Context, ba roachpb.BatchRequest) (*roachpb.B
 	if ba.IsAdmin() {
 		log.Trace(ctx, "admin path")
 		br, pErr = r.addAdminCmd(ctx, ba)
-	} else if ba.IsReadOnly() {
-		log.Trace(ctx, "read-only path")
-		br, pErr = r.addReadOnlyCmd(ctx, ba)
 	} else if ba.IsWrite() {
 		log.Trace(ctx, "read-write path")
 		br, pErr = r.addWriteCmd(ctx, ba, nil)
-	} else if len(ba.Requests) == 0 {
+	} else if len(ba.Requests) > 0 {
```

Comment:
Seems a little sloppy here. I know this is what the implementation of IsReadOnly does as well, but still it seems cleaner to call that. BTW, a more "efficient" micro optimization would be moving `IsAdmin` to the last branch because that rarely ever fires, but has to go through the whole batch.

On the grander scheme, we have a lot of places that comb the whole batch and caching some of that information in the header could be useful (but it's frail because we change the requests in a bunch of places).
Most of those should be really micro, so not worth it at the moment.